src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -35,6 +35,9 @@ module FeatureLoader
     # A snapshot of $LOADED_FEATURES, to check if the @loaded_features_index cache is up to date.
     @loaded_features_version = -1
 
+    @features_realpath_cache = {}
```

Comment:
I don't think this cache is correct, for instance it seems broken if CWD changes or $LOAD_PATH changes.
Probably that part shouldn't be cached.

You could cache absolute_path->realpath though.